src/sshfs_mount/sshfs_server.cpp

```diff
@@ -0,0 +1,125 @@
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <QStringList>
+
+#include "../ssh/ssh_client_key_provider.h" // FIXME
+#include <multipass/exceptions/sshfs_missing_error.h>
+#include <multipass/logging/log.h>
+#include <multipass/logging/standard_logger.h>
+#include <multipass/ssh/ssh_session.h>
+#include <multipass/sshfs_mount/sshfs_mount.h>
+
+#include <signal.h>
+#include <sys/prctl.h>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+using namespace std;
+
+namespace
+{
+unordered_map<int, int> deserialise_id_map(const char* in)
+{
+    unordered_map<int, int> id_map;
+    QString input(in);
+    auto maps = input.split(',', QString::SkipEmptyParts);
+    for (auto map : maps)
+    {
+        auto ids = map.split(":");
+        if (ids.count() != 2)
+        {
+            cerr << "Incorrect ID mapping syntax";
+            continue;
+        }
+        bool ok1, ok2;
+        int from = ids.first().toInt(&ok1);
+        int to = ids.last().toInt(&ok2);
+        if (!ok1 || !ok2)
+        {
+            cerr << "Incorrect ID mapping ids found, ignored" << endl;
+            continue;
+        }
+        id_map.insert({from, to});
+    }
+    return id_map;
+}
+} // namespace
+
+int main(int argc, char* argv[])
+{
+    prctl(PR_SET_PDEATHSIG, SIGHUP); // ensure if parent dies, this process gets the SIGHUP signal
+
+    if (argc != 8)
+    {
+        cerr << "Incorrect arguments" << endl;
+        exit(2);
+    }
+
+    const auto key = getenv("KEY");
+    if (key == nullptr)
+    {
+        cerr << "KEY not set" << endl;
+        exit(2);
+    }
+    const auto priv_key_blob = string(key);
+    const auto host = string(argv[1]);
+    const int port = atoi(argv[2]);
+    const auto username = string(argv[3]);
+    const auto source_path = string(argv[4]);
+    const auto target_path = string(argv[5]);
+    const unordered_map<int, int> gid_map = deserialise_id_map(argv[6]);
+    const unordered_map<int, int> uid_map = deserialise_id_map(argv[7]);
+
+    auto logger = std::make_shared<mpl::StandardLogger>(mpl::Level::error); // QUESTION - how to pass verbosity level?
+    mpl::set_logger(logger);
+
+    try
+    {
+        mp::SSHSession session{host, port, username, mp::SSHClientKeyProvider{priv_key_blob}};
+
+        mp::SshfsMount sshfs_mount(move(session), source_path, target_path, gid_map, uid_map);
+
+        // ssh lives on its own thread, use this thread to listen for quit signal
+        sigset_t sigset;
```

Comment:
I don't think this and the following 8 lines are portable, so need to move into platform specific parts and hopefully there is a Windows equivalent.

Reply:
IIRC this does actually compile on windows, but doesn't really do anything.

Reply:
Do we need to do equivalent functionality on Windows to achieve the same thing?  If so, then I'd really prefer this to be moved to the platform abstractions.  Also, even if this compiles on Windows but is essentially a no-op, I think it's kind of messy to leave it there.

Reply:
This can now be handled by any Windows platform specific bits.